pkg/querier/queryrange/views.go

```diff
@@ -0,0 +1,332 @@
+package queryrange
```

Comment:
Nit: this file is rather large and very complex, we might consider splitting it up.

Reply:
I thought about this but wanted to make it easy to remove. It's less than 400 lines. `codec.go` is 1300+. I'm happy to split it into each view though.

Reply:
We could remove the whole directory just as easily though?